tests/Unit/NumericalAlgorithms/DiscontinuousGalerkin/Test_MetricIdentityJacobian.cpp

```diff
@@ -0,0 +1,160 @@
+
+#include "DataStructures/ApplyMatrices.hpp"
+#include "DataStructures/DataVector.hpp"
+#include "DataStructures/Matrix.hpp"
+#include "DataStructures/Tensor/EagerMath/Determinant.hpp"
+#include "DataStructures/Tensor/Tensor.hpp"
+#include "Domain/CoordinateMaps/Affine.hpp"
+#include "Domain/CoordinateMaps/CoordinateMap.hpp"
+#include "Domain/CoordinateMaps/CoordinateMap.tpp"
+#include "Domain/CoordinateMaps/ProductMaps.hpp"
+#include "Domain/CoordinateMaps/ProductMaps.tpp"
+#include "Domain/CoordinateMaps/Wedge2D.hpp"
+#include "Domain/LogicalCoordinates.hpp"
+#include "Framework/TestHelpers.hpp"
+#include "Helpers/DataStructures/MakeWithRandomValues.hpp"
+#include "NumericalAlgorithms/DiscontinuousGalerkin/MetricIdentityJacobian.hpp"
+#include "NumericalAlgorithms/Spectral/Mesh.hpp"
+#include "NumericalAlgorithms/Spectral/Spectral.hpp"
+#include "Utilities/Gsl.hpp"
+#include "Utilities/MakeArray.hpp"
+
+namespace {
+using Affine = domain::CoordinateMaps::Affine;
+using Affine2D = domain::CoordinateMaps::ProductOf2Maps<Affine, Affine>;
+using Affine3D = domain::CoordinateMaps::ProductOf3Maps<Affine, Affine, Affine>;
+
+template <size_t VolumeDim>
+auto make_map() noexcept;
+
+template <>
+auto make_map<1>() noexcept {
+  // Works when identity...
+  return domain::make_coordinate_map<Frame::Logical, Frame::Inertial>(
+      Affine{-1.0, 1.0, -0.3, 0.7});
+}
+
+template <>
+auto make_map<2>() noexcept {
+  return domain::make_coordinate_map<Frame::Logical, Frame::Inertial>(
+      Affine2D{{-1.0, 1.0, -1.0, -0.99}, {-1.0, 1.0, -1.0, -0.99}},
+      domain::CoordinateMaps::Wedge2D{1.0, 2.0, 0.0, 1.0, {}, false});
+}
+
+template <>
+auto make_map<3>() noexcept {
+  return domain::make_coordinate_map<Frame::Logical, Frame::Inertial>(
+      Affine3D{{-1.0, 1.0, -1.0, -0.99},
+               {-1.0, 1.0, -1.0, -0.99},
+               {-1.0, 1.0, -1.0, 1.0}},
+      domain::CoordinateMaps::ProductOf2Maps<domain::CoordinateMaps::Wedge2D,
+                                             Affine>{
+          {1.0, 2.0, 0.0, 1.0, {}, false}, {0.0, 1.0, 0.0, 1.0}});
+}
+
+
+template <size_t Dim>
+void test(const Mesh<Dim>& mesh) {
+  CAPTURE(Dim);
+  CAPTURE(mesh);
+  std::uniform_real_distribution<double> dist(-1.0, 2.3);
+  MAKE_GENERATOR(gen);
+  tnsr::I<DataVector, Dim, Frame::Inertial> inertial_coords{
+      mesh.number_of_grid_points()};
+  Jacobian<DataVector, Dim, Frame::Logical, Frame::Inertial> jacobian{
+      mesh.number_of_grid_points()};
+
+  fill_with_random_values(make_not_null(&inertial_coords), make_not_null(&gen),
+                          make_not_null(&dist));
+  fill_with_random_values(make_not_null(&jacobian), make_not_null(&gen),
+                          make_not_null(&dist));
+
+  InverseJacobian<DataVector, Dim, Frame::Logical, Frame::Inertial> result{};
+
+  dg::metric_identity_jacobian(make_not_null(&result), mesh, inertial_coords,
+                               jacobian);
+
+  // To test, take numerical divergence.
```

Comment:
How about "Check metric identities are satisfied by taking the numerical divergence"? I think that is clearer about what you are actually testing.